File: 백준/Bronze/6246. 풍선 놀이/풍선 놀이.py
n, q = map(int, input().split())
arr = [0] * (n + 1)
for _ in range(q):
    L, I = map(int, input().split())
    for i in range(L, n + 1, I):
        arr[i] = 1
        
print(n - sum(arr[1:]))

File: 백준/Bronze/6246. 풍선 놀이/풍선 놀이.java
import java.util.Scanner;

public class Main {
    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);
        int n = scanner.nextInt();
        int q = scanner.nextInt();
        int[] arr = new int[n + 1];
        
        for (int i = 0; i < q; i++) {
            int L = scanner.nextInt();
            int I = scanner.nextInt();
            
            for (int j = L; j <= n; j += I) {
                arr[j] = 1;
            }
        }
        
        int sum = 0;
        for (int i = 1; i <= n; i++) {
            sum += arr[i];
        }
        
        System.out.println(n - sum);
    }
}